xarray/tests/test_coding.py

```diff
@@ -431,7 +514,26 @@ def test_infer_datetime_units(self):
                                 (pd.to_datetime(['NaT']),
                                  'days since 1970-01-01 00:00:00'),
                                 ]:
-            self.assertEqual(expected, conventions.infer_datetime_units(dates))
+            self.assertEqual(expected, coding.infer_datetime_units(dates))
+
+    @requires_netCDF4
+    def test_infer_netcdftime_datetime_units(self):
```

Comment:
Note that while the logic I used in [`infer_datetime_units`](https://github.com/pydata/xarray/pull/1252/files#diff-796786eab6c7f79e802c9f4617ef7059R212) is basically the same for `netcdftime` datetimes and standard datetimes, the behavior in these test cases is slightly different.  I think this may have something to do with the fact that timedeltas are not always exact to microsecond-precision using `netcdftime` objects.  For example I would expect this to return exactly 3600 seconds:
```
>>> DatetimeNoLeap(1900, 1, 1, 13) - DatetimeNoLeap(1900, 1, 1, 12)
datetime.timedelta(0, 3599, 999985)
```
In the case of ordinary datetimes it does:
```
>>> datetime(1900, 1, 1, 13) - datetime(1900, 1, 1, 12)
datetime.timedelta(0, 3600)
```
This is probably not something we can address in xarray (maybe upstream?), but I just wanted to clarify why the results of these tests are different.